martin/src/config/file/main.rs

```diff
@@ -365,6 +393,45 @@ impl Config {
     }
 }
 
+/// Describes the action to take during startup when configuration is found to be invalid
+/// but Martin could still startup in a degraded state (ie, some sources not served).
+#[derive(PartialEq, Eq, Debug, Clone, Copy, Default, Serialize, Deserialize, ValueEnum)]
+#[serde(rename_all = "lowercase")]
+pub enum OnInvalid {
+    /// Log warning messages, abort if the error is critical
+    Warn,
+    /// Log warnings as errors, abort startup
+    #[default]
+    Abort,
+}
+
+impl OnInvalid {
+    /// Handle warnings based on `policy`
+    pub fn handle_warnings(policy: Self, warnings: &[TileSourceWarning]) -> MartinResult<()> {
+        if warnings.is_empty() {
+            return Ok(());
+        }
+        let level = match policy {
+            OnInvalid::Warn => Level::Warn,
+            OnInvalid::Abort => Level::Error,
+        };
+        if warnings.len() == 1 {
+            let warning = &warnings[0];
+            log!(level, "Tile source resolution warning: {warning}");
+        } else {
+            log!(level, "Tile source resolution warnings:");
+            for warning in warnings {
+                log!(level, "  - {warning}");
+            }
+        }
+
+        match policy {
+            OnInvalid::Abort => Err(MartinError::TileResolutionWarningsIssued),
+            OnInvalid::Warn => Ok(()),
+        }
+    }
```

Comment:
nit: lets change this to the imo more idiomatic `policy.handle_tile_warnings(warnings)` and switch the way warnings are matched.

```suggestion
    pub fn handle_tile_warnings(self, warnings: &[TileSourceWarning]) -> MartinResult<()> {
        if warnings.is_empty() {
            return Ok(());
        }
        let level = match self {
            OnInvalid::Warn => Level::Warn,
            OnInvalid::Abort => Level::Error,
        };
        match warnings {
            [warning] => log!(level, "Tile source resolution warning: {warning}"),
            warnings => {
                log!(level, "Tile source resolution warnings:");
                for warning in warnings {
                    log!(level, "  - {warning}");
                }
            }
        }

        match self {
            OnInvalid::Abort => Err(MartinError::TileResolutionWarningsIssued),
            OnInvalid::Warn => Ok(()),
        }
    }
```